syne_tune/optimizer/schedulers/multiobjective/nsga2_searcher.py

```diff
@@ -0,0 +1,177 @@
+                vars[hp_name] = PyMOOInteger(bounds=(0, hp.size - 1))
+            elif isinstance(hp, Float):
+                vars[hp_name] = Real(bounds=(hp.lower, hp.upper))
+
+        super().__init__(vars=vars, n_obj=n_obj, n_ieq_constr=0, **kwargs)
+
+
+class NSGA2Searcher(StochasticSearcher):
+    """
+    This is a wrapper around the NSGA-2 [1] implementation of pymoo [2].
+
+    [1] K. Deb, A. Pratap, S. Agarwal, and T. Meyarivan.
+    A fast and elitist multiobjective genetic algorithm: nsga-II.
+    Trans. Evol. Comp, 6(2):182–197, April 2002.
+
+    [2] J. Blank and K. Deb
+    pymoo: Multi-Objective Optimization in Python
+    IEEE Access, 2020
+
+    :param config_space: Configuration space
+    :param metric: Name of metric passed to :meth:`~update`. Can be obtained from
+        scheduler in :meth:`~configure_scheduler`. In the case of multi-objective optimization,
+         metric is a list of strings specifying all objectives to be optimized.
+    :param points_to_evaluate: List of configurations to be evaluated
+        initially (in that order). Each config in the list can be partially
+        specified, or even be an empty dict. For each hyperparameter not
+        specified, the default value is determined using a midpoint heuristic.
+        If ``None`` (default), this is mapped to ``[dict()]``, a single default config
+        determined by the midpoint heuristic. If ``[]`` (empty list), no initial
+        configurations are specified.
+    :param mode: Should metric be minimized ("min", default) or maximized
+        ("max"). In the case of multi-objective optimization, mode can be a list defining for
+        each metric if it is minimized or maximized
+    :param population_size: Size of the population
+    """
+
+    def __init__(
+        self,
+        config_space: Dict[str, Any],
+        metric: List[str],
+        mode: Union[List[str], str],
+        points_to_evaluate: Optional[List[dict]] = None,
+        population_size: int = 20,
+        **kwargs,
+    ):
+        super(NSGA2Searcher, self).__init__(
+            config_space, metric, points_to_evaluate=points_to_evaluate, **kwargs
+        )
+        if isinstance(mode, str):
+            self._mode = [mode] * len(metric)
+        else:
+            self._mode = mode
+
+        self.hp_names = []
+        for hp_name, hp in config_space.items():
+            if isinstance(hp, Domain):
+                self.hp_names.append(hp_name)
+                assert type(hp) in [
+                    Categorical,
+                    Integer,
+                    Float,
+                    FiniteRange,
+                ], f"Type {type(hp)} for hyperparameter {hp_name} not supported."
+
+        self.problem = MultiObjectiveMixedVariableProblem(
+            config_space=config_space, n_var=len(self.hp_names), n_obj=len(metric)
+        )
+        self.algorithm = NSGA2(
+            pop_size=population_size,
+            sampling=MixedVariableSampling(),
```

Comment:
Is there any chance we can control the random seed of these components? If so, please try to do that, and use `self.random_seed`.

Otherwise, we lose the feature that results are repeatable, which we should retain for methods that just do random sampling.